app/assets/javascripts/models/data/project.js.coffee

```diff
@@ -87,9 +87,12 @@ ns.Project = do (ko,
       options = @serviceOptions.serialize()
       $(document).trigger('dataaccessevent', [@collection.id, options])
 
+      form_data = []
+      form_data.push method.form for method in @granuleAccessOptions().methods
       id: @collection.id
       params: param(@collection.granuleDatasource()?.toQueryParams() ? @collection.query.globalParams())
       serviceOptions: options
+      form: form_data
```

Comment:
No need to do any of that. Have the server tell the client what the sha1 is, then have the client pass the sha1 back to the server. The client doesn't need to compute the sha1.